plugins/ltac/extratactics.mlg

```diff
@@ -450,7 +448,7 @@ END
 (* Subst                                                              *)
 
 TACTIC EXTEND subst
-| [ "subst" ne_var_list(l) ] -> { subst l }
+| [ "subst" ne_hyp_list(l) ] -> { subst l }
```

Comment:
Doesn't subst take variables rather than hypotheses?  I don't think we'd want the doc to say "hyp" here if it's not a hypothesis.  At the moment, the edited doc grammar uses `ident` here, but we've not applied those changes in the the tactics chapter yet.

Maybe should make hyp and var synonyms without deprecating?

Reply:
That's actually the big question! Internally, definition=theorem, body=proof, variable=hypothesis, parameter=axiom, recursion=induction, ... that's the whole issue of documenting a unique syntax which has a double meaning. This leads to a tension between providing intuitive names and avoiding redundancy and this is a major issue in the pedagogy of Coq.

I'm afraid that if we really want to maintain a difference of vocabulary, the difference would have to be kept in the structure of terms themselves, so that this difference (decided by the user) is consistently used for parsing and printing. Why not? But on the other side, I tend to think that users, after some while, would themselves get use to the identification of concepts and stop making a formal difference (as most Coq users do after some time, where we freely switch between the programming vocabulary and the logic vocabulary, meaning either of them).

So, shortly, this is indeed a critical pedagogical issue.